.github/workflows/publish.yml

```diff
@@ -14,10 +14,16 @@ jobs:
         python-version: '3.7'
     - name: Build wheels
       run: ./scripts/build.sh
+    - name: Publish to TestPyPI
+      env:
+        TWINE_USERNAME: ${{ secrets.test_pypi_username }}
+        TWINE_PASSWORD: ${{ secrets.test_pypi_password }}
+      run: |
+        pip install twine
```

Comment:
should installing deps/tools be an independent step before other steps that depend on it?

Reply:
Good point, done.